### PR TITLE
Changes PullRequest Index to ForgeRemoteID type

### PR DESCRIPTION
**Pull request by mzampetakis**
This allows using string as pull requst ID in the forges. Also, provides a uniform type for remote forge IDs among different objects.

Resolves #2746 
Needed by #2355

### CODE REVIEW

File: server/forge/bitbucket/bitbucket.go [hunk 1]
```diff
@@ -369,7 +370,7 @@ func (c *config) PullRequests(ctx context.Context, u *model.User, r *model.Repo,
 	result := []*model.PullRequest{}
 	for _, pullRequest := range pullRequests {
 		result = append(result, &model.PullRequest{
-			Index: int64(pullRequest.ID),
+			Index: model.ForgeRemoteID(strconv.Itoa(int(pullRequest.ID))),
```
Comment (6543) [inline server/forge/bitbucket/bitbucket.go:373]: Do we need int conversion first?

File: server/forge/github/github.go [hunk 2]
```diff
@@ -296,7 +296,7 @@ func (c *client) PullRequests(ctx context.Context, u *model.User, r *model.Repo,
 	result := make([]*model.PullRequest, len(pullRequests))
 	for i := range pullRequests {
 		result[i] = &model.PullRequest{
-			Index: int64(pullRequests[i].GetNumber()),
+			Index: model.ForgeRemoteID(strconv.Itoa((pullRequests[i].GetNumber()))),
```
[the remark this answers is not in the document]
Comment (mzampetakis) [reply server/forge/github/github.go:299]: Definitely not! Thanks!

File: server/forge/gitlab/gitlab.go [hunk 3]
```diff
@@ -326,7 +326,7 @@ func (g *GitLab) PullRequests(ctx context.Context, u *model.User, r *model.Repo,
 	result := make([]*model.PullRequest, len(pullRequests))
 	for i := range pullRequests {
 		result[i] = &model.PullRequest{
-			Index: int64(pullRequests[i].ID),
+			Index: model.ForgeRemoteID(strconv.Itoa((pullRequests[i].ID))),
```
[the remark this answers is not in the document]
Comment (mzampetakis) [reply server/forge/gitlab/gitlab.go:329]: As before. Thanks!

File: server/model/pull_request.go [hunk 4]
```diff
@@ -15,6 +15,6 @@
 package model
 
 type PullRequest struct {
-	Index int64  `json:"index"`
-	Title string `json:"title"`
+	Index ForgeRemoteID `json:"index"`
```
[the remark this answers is not in the document]
Comment (mzampetakis) [reply server/model/pull_request.go:18]: As far as I can see, PRs are not stored in the database at all. So there is no such case.
However, I am not sure if the endpoint `/repos/{repo_id}/pull_requests [get]` is used by the UI in order to get updated accordingly. Swagger seems seems that will be generated automatically from the sources.

[the remark this answers is not in the document]
Comment (mzampetakis) [reply server/model/pull_request.go:18]: Should I run it at the PR, or will it run automatically by a pipeline?

**Comment by 6543** [comment]
PS: we are currently in feature freeze, until v2.0.0 is tagged :)

File: server/model/pull_request.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (mzampetakis) [reply server/model/pull_request.go:18]: OK, I run the command which updates the docs!

**Comment by mzampetakis** [comment]
> PS: we are currently in feature freeze, until v2.0.0 is tagged :)

Yes, I am aware about it and looking forward for the release!  🥳 
This can definitely wait!

**Review by anbraten — APPROVED** [review]

**Pull request merged**